src/cartesian/XAxis.tsx

```diff
@@ -31,7 +36,24 @@ interface XAxisProps extends BaseAxisProps {
 
 export type Props = Omit<SVGProps<SVGElement>, 'scale'> & XAxisProps;
 
-export const XAxis: FunctionComponent<Props> = () => null;
+export const XAxis: FunctionComponent<Props> = ({ xAxisId }: Props) => {
+  const viewBox = useViewBox();
+  const axisOptions = useXAxisOrThrow(xAxisId);
+
+  if (axisOptions == null) {
+    return null;
+  }
+
+  return (
+    // @ts-expect-error the axisOptions type is not exactly what CartesianAxis is expecting.
```

Comment:
why/what are the consequences of this?

Reply:
Why? This is exactly the same type hole as in the original method here: https://github.com/recharts/recharts/blob/master/src/chart/generateCategoricalChart.tsx#L1859 except in there it's hidden behind `as any`.

The consequence? Recharts axis types need some more love. When we're done with the context refactor, it will allow us to simplify and remove lot of extra code.

Reply:
Thanks!